hugr/src/hugr/serialize/test.rs

```diff
@@ -349,79 +367,34 @@ fn serialize_types_roundtrip() {
+    use proptest::prelude::*;
+
+    proptest! {
+        #[test]
+        fn prop_roundtrip_type(t:  Type) {
+            check_testing_roundtrip(t)
+        }
 
-fn polyfunctype1() -> PolyFuncType {
-    let mut extension_set = ExtensionSet::new();
-    extension_set.insert_type_var(1);
-    let function_type = FunctionType::new_endo(type_row![]).with_extension_delta(extension_set);
-    PolyFuncType::new([TypeParam::max_nat(), TypeParam::Extensions], function_type)
-}
+        #[test]
+        fn prop_roundtrip_poly_func_type(t: PolyFuncType) {
+            check_testing_roundtrip(t)
+        }
 
-#[rstest]
-#[case(FunctionType::new_endo(type_row![]).into())]
-#[case(polyfunctype1())]
-#[case(PolyFuncType::new([TypeParam::Opaque { ty: int_custom_type(TypeArg::BoundedNat { n: 1 }) }], FunctionType::new_endo(type_row![Type::new_var_use(0, TypeBound::Copyable)])))]
-#[case(PolyFuncType::new([TypeBound::Eq.into()], FunctionType::new_endo(type_row![Type::new_var_use(0, TypeBound::Eq)])))]
-#[case(PolyFuncType::new([TypeParam::List { param: Box::new(TypeBound::Any.into()) }], FunctionType::new_endo(type_row![])))]
-#[case(PolyFuncType::new([TypeParam::Tuple { params: [TypeBound::Any.into(), TypeParam::bounded_nat(2.try_into().unwrap())].into() }], FunctionType::new_endo(type_row![])))]
-fn roundtrip_polyfunctype(#[case] poly_func_type: PolyFuncType) {
-    check_testing_roundtrip(poly_func_type)
-}
+        #[test]
+        fn prop_roundtrip_value(t: Value) {
+            check_testing_roundtrip(t)
+        }
 
-#[rstest]
-#[case(ops::Module)]
-#[case(ops::FuncDefn { name: "polyfunc1".into(), signature: polyfunctype1()})]
-#[case(ops::FuncDecl { name: "polyfunc2".into(), signature: polyfunctype1()})]
-#[case(ops::AliasDefn { name: "aliasdefn".into(), definition: Type::new_unit_sum(4)})]
-#[case(ops::AliasDecl { name: "aliasdecl".into(), bound: TypeBound::Any})]
-#[case(ops::Const::new(Value::false_val()))]
-#[case(ops::Const::new(Value::function(crate::builder::test::simple_dfg_hugr()).unwrap()))]
-#[case(ops::Input::new(type_row![Type::new_var_use(3,TypeBound::Eq)]))]
-#[case(ops::Output::new(vec![Type::new_function(FunctionType::new_endo(type_row![]))]))]
-#[case(ops::Call::try_new(polyfunctype1(), [TypeArg::BoundedNat{n: 1}, TypeArg::Extensions{ es: ExtensionSet::singleton(&PRELUDE_ID)} ], &EMPTY_REG).unwrap())]
-#[case(ops::CallIndirect { signature : FunctionType::new_endo(type_row![BOOL_T]) })]
-fn roundtrip_optype(#[case] optype: impl Into<OpType> + std::fmt::Debug) {
-    check_testing_roundtrip(NodeSer {
-        parent: portgraph::NodeIndex::new(0).into(),
-        input_extensions: None,
-        op: optype.into(),
-    });
+        #[test]
+        fn prop_roundtrip_optype(op in ((0..(std::u32::MAX / 2) as usize). prop_map(|x| portgraph::NodeIndex::new(x).into()), any::<Option<ExtensionSet>>(), any::<OpType>()).prop_map(|(parent, input_extensions, op)| NodeSer { parent, input_extensions, op })) {
```

Comment:
This is a little hard to read...is it possible to outline this `NodeSer`-factory? I think the overall is `fn prop_roundtrip_optype(op in <thing>)` which is presumably a proptest-specific construct, but maybe that can be an `Iterator<NodeSer>` or something? (A shame to have to construct `NodeSer`s really - ideally `prop_roundtrip_optype` should take an Optype, but I guess that doesn't fit the serialization format?)